### PR TITLE
Migration guide

### LINKED ISSUE #297: Migration instructions coming from TF cloud
TF cloud doesn't make it easy to migrate states to other backends.

There is https://developer.hashicorp.com/terraform/tutorials/cloud/cloud-state-api#download-the-state-file, yet many people would probably appreciate a more "simple" approach to do this.

As I've just gone through this, I've created a gist to simplify the process: https://gist.github.com/pat-s/6cc1a931866e869f1adb1c84f715ef92. Feel free to adapt/reuse, just wanted to drop it here for future people to get up to speed quickly!

### PR DESCRIPTION
**Pull request by leg100**
Fixes #297 

@pat-s Can you give this a proof-read and try it out? Thanks.

### CODE REVIEW

**Comment by pat-s** [comment]
Will do soon!

File: docs/tfc_migration.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+			hostname = "app.terraform.io"
+			organization = "automatize"
+
+			workspaces {
+			  name = "my-workspace"
+			}
+		  }
+		}
+
+1. To migrate to OTF you only need to update the hostname:
```
Comment (pat-s) [inline docs/tfc_migration.md:65]: Did that really work for you? 😮 
I tried this and always got rejected with "migrating form 'cloud' to other backends is not supported right now'. 

That's why I went with the cumbersome API workaround I shared some time ago. Did something change on the terraform side?

I mean great if it works that easily now, just wondering 😅

Comment (leg100) [reply docs/tfc_migration.md:65]: That error only occurs when using the `cloud` block, not the `remote` backend. Hence the more long-winded instructions for migrating from the `cloud` block.

**Comment by pat-s** [comment]
> But I tried again the other day, prior to writing this document, and this time I found you could migrate away from the cloud block, using the steps I documented. I'm not sure why, or whether the exact steps are different. But it worked for me.

I see. Maybe something also changed upstream, who knows. 

> I would appreciate it if you could try again using the documented steps. If you could setup a workspace in TFC using the cloud block and follow the document to migrate to OTF.

I can confirm it works following your steps. Terraform 1.4.6. 

In the end all that counts is that it works now in a somewhat "easy" way. That is great! I am not too motivated finding out what has changed or what I did wrong a few weeks ago. So let's move on and if there are more users having issues, they will report anyhow 🙂️

**Pull request merged**